### PR TITLE
PHP unit test database connection null side effect

### PR DESCRIPTION
**Pull request by PVince81**
@DeepDiver1975 see the two test classes `CustomGroupsBackendTest` and `CustomGroupsManagerTest`.

- CustomGroupsBackendTest: this one requires no DB calls as it only use a mock object, so I removed `@group DB` from it
- CustomGroupsManagerTest: requires DB and has `@group DB` defined, but the DB connection is null !

If you add back `@group DB to the first class, then the second class suddenly receives a proper connection.

Possibly a bug in the `TestCase` magic in core related to forbidding DB access.

### CODE REVIEW

**Comment by PVince81** [comment]
When you run the tests with `make test-php` (or `cd tests/unit; phpunit`):
```
2) OCA\CustomGroups\Tests\unit\CustomGroupsManagerTest::testCreateGroup
TypeError: Argument 1 passed to OCA\CustomGroups\CustomGroupsManager::__construct() must implement interface OCP\IDBConnection, null given, called in /srv/www/htdocs/owncloud/apps3/customgroups/tests/unit/CustomGroupsManagerTest.php on line 47

/srv/www/htdocs/owncloud/apps3/customgroups/lib/CustomGroupsManager.php:34
/srv/www/htdocs/owncloud/apps3/customgroups/tests/unit/CustomGroupsManagerTest.php:47
/home/vincent/.composer/vendor/phpunit/phpunit/src/TextUI/Command.php:149
/home/vincent/.composer/vendor/phpunit/phpunit/src/TextUI/Command.php:100

3) OCA\CustomGroups\Tests\unit\CustomGroupsManagerTest::testCreateGroupDuplicateUri
TypeError: Argument 1 passed to OCA\CustomGroups\CustomGroupsManager::__construct() must implement interface OCP\IDBConnection, null given, called in /srv/www/htdocs/owncloud/apps3/customgroups/tests/unit/CustomGroupsManagerTest.php on line 47

/srv/www/htdocs/owncloud/apps3/customgroups/lib/CustomGroupsManager.php:34
/srv/www/htdocs/owncloud/apps3/customgroups/tests/unit/CustomGroupsManagerTest.php:47
/home/vincent/.composer/vendor/phpunit/phpunit/src/TextUI/Command.php:149
/home/vincent/.composer/vendor/phpunit/phpunit/src/TextUI/Command.php:100
```

**Comment by PVince81** [comment]
Raised in core https://github.com/owncloud/core/issues/26771, likely a bug in `\Test\TestCase`

File: tests/unit/CustomGroupsBackendTest.php [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+ */
+class CustomGroupsBackendTest extends \Test\TestCase {
+	const GROUP_ID_PREFIX = CustomGroupsBackend::GROUP_ID_PREFIX;
+
+	/**
+	 * @var CustomGroupsManager
+	 */
+	private $manager;
+
+	public function setUp() {
```
Comment (DeepDiver1975) [inline tests/unit/CustomGroupsBackendTest.php]: you have to call parent::setup()

**Comment by PVince81** [comment]
Will do in the main PR, thanks

**Pull request closed without merging**